lib/ret/http_utils.ex

```diff
@@ -1,31 +1,46 @@
 defmodule Ret.HttpUtils do
   use Retry
 
-  def retry_head_until_success(url, headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000),
-    do: retry_until_success(:head, url, "", headers, cap_ms, expiry_ms)
+  def retry_head_until_success(url, options \\ []),
+    do: retry_until_success(:head, url, options)
 
-  def retry_get_until_success(url, headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000),
-    do: retry_until_success(:get, url, "", headers, cap_ms, expiry_ms)
+  def retry_get_until_success(url, options \\ []),
+    do: retry_until_success(:get, url, options)
 
-  def retry_post_until_success(url, body, headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000),
-    do: retry_until_success(:post, url, body, headers, cap_ms, expiry_ms)
+  def retry_post_until_success(url, body, options \\ []),
+    do: retry_until_success(:post, url, Keyword.merge(options, body: body))
 
-  def retry_put_until_success(url, body, headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000),
-    do: retry_until_success(:put, url, body, headers, cap_ms, expiry_ms)
+  def retry_put_until_success(url, body, options \\ []),
+    do: retry_until_success(:put, url, Keyword.merge(options, body: body))
```

Comment:
These wrappers are convenient in cases where we like to write `url |> retry_foo_until_success(options)` Is it also convenient for `body` to be a top level argument here instead of requiring the call site to merge it into `options`?

Reply:
Ah, yeah, that would be a bit cleaner. Good call